File: replication/src/core/com/db4o/foundation/ObjectSetAbstractFacade.java
/* Copyright (C) 2004 - 2005  db4objects Inc.  http://www.db4o.com */

package com.db4o.foundation;

import java.util.*;

import com.db4o.*;
import com.db4o.ext.*;

public abstract class ObjectSetAbstractFacade implements ObjectSet{
    
    public abstract boolean hasNext();

    public abstract Object next();

    public ExtObjectSet ext() {
        throw notSupported();
    }

    public void reset() {
        throw notSupported();
    }

    public int size() {
        throw notSupported();
    }
    
    private final RuntimeException notSupported(){
        return new UnsupportedOperationException("not supported");
    }

    public boolean isEmpty() {
        throw notSupported();
    }

    public boolean contains(Object o) {
        throw notSupported();
    }

    public Iterator iterator() {
        throw notSupported();
    }

    public Object[] toArray() {
        throw notSupported();
    }

    public Object[] toArray(Object[] arg0) {
        throw notSupported();
    }

    public boolean add(Object arg0) {
        throw notSupported();
    }

    public boolean remove(Object o) {
        throw notSupported();
    }

    public boolean containsAll(Collection arg0) {
        throw notSupported();
    }

    public boolean addAll(Collection arg0) {
        throw notSupported();
    }

    public boolean addAll(int arg0, Collection arg1) {
        throw notSupported();
    }

    public boolean removeAll(Collection arg0) {
        throw notSupported();
    }

    public boolean retainAll(Collection arg0) {
        throw notSupported();
    }

    public void clear() {
        throw notSupported();
    }

    public Object get(int index) {
        throw notSupported();
    }

    public Object set(int arg0, Object arg1) {
        throw notSupported();
    }

    public void add(int arg0, Object arg1) {
        throw notSupported();
    }

    public Object remove(int index) {
        throw notSupported();
    }

    public int indexOf(Object o) {
        throw notSupported();
    }

    public int lastIndexOf(Object o) {
        throw notSupported();
    }

    public ListIterator listIterator() {
        throw notSupported();
    }

    public ListIterator listIterator(int index) {
        throw notSupported();
    }

    public List subList(int fromIndex, int toIndex) {
        throw notSupported();
    }

    public void remove() {
        throw notSupported();
    }


}

File: drs/drs.net/in/Db4objects/Db4o/Drs/Foundation/ObjectSetAbstractFacade.cs
using System;
using System.Collections;
using Db4objects.Db4o.Ext;

namespace Db4objects.Db4o.Drs.Foundation
{
	public abstract class ObjectSetAbstractFacade : Db4objects.Db4o.IObjectSet
	{
		public IExtObjectSet Ext()
		{
			throw new NotImplementedException();
		}

		public bool HasNext()
		{
			throw new NotImplementedException();
		}

		public object Next()
		{
			throw new NotImplementedException();
		}

		public void Reset()
		{
			throw new NotImplementedException();
		}

		public int Size()
		{
			throw new NotImplementedException();
		}

		public int Add(object value)
		{
			throw new NotImplementedException();
		}

		public bool Contains(object value)
		{
			throw new NotImplementedException();
		}

		public void Clear()
		{
			throw new NotImplementedException();
		}

		public int IndexOf(object value)
		{
			throw new NotImplementedException();
		}

		public void Insert(int index, object value)
		{
			throw new NotImplementedException();
		}

		public void Remove(object value)
		{
			throw new NotImplementedException();
		}

		public void RemoveAt(int index)
		{
			throw new NotImplementedException();
		}

		public object this[int index]
		{
			get { throw new NotImplementedException(); }
			set { throw new NotImplementedException(); }
		}

		public bool IsReadOnly
		{
			get { throw new NotImplementedException(); }
		}

		public bool IsFixedSize
		{
			get { throw new NotImplementedException(); }
		}

		public void CopyTo(Array array, int index)
		{
			throw new NotImplementedException();
		}

		public int Count
		{
			get { throw new NotImplementedException(); }
		}

		public object SyncRoot
		{
			get { throw new NotImplementedException(); }
		}

		public bool IsSynchronized
		{
			get { throw new NotImplementedException(); }
		}

		public IEnumerator GetEnumerator()
		{
			throw new NotImplementedException();
		}
	}
}